ledger/catchpointtracker.go

```diff
@@ -1446,21 +1442,55 @@ func accountHashBuilderV6(addr basics.Address, accountData *baseAccountData, enc
 	return finishV6(hash, prehash)
 }
 
-// accountHashBuilderV6 calculates the hash key used for the trie by combining the account address and the account data
-func resourcesHashBuilderV6(addr basics.Address, cidx basics.CreatableIndex, ctype basics.CreatableType, updateRound uint64, encodedResourceData []byte) []byte {
-	hash := hashBufV6(updateRound, byte(ctype+1)) // one or two ( asset / application ) so we could differentiate the hashes.
+// hashKind enumerates the possible data types hashed into a catchpoint merkle
+// trie.  Each merkle trie hash includes the hashKind byte at a known-offset.
+// By encoding hashKind at a known-offset, it's possible for hash readers to
+// disambiguate the hashed resource.
+//go:generate stringer -type=hashKind
+type hashKind byte
+
+// Defines known kinds of hashes.  Changing an enum ordinal value is a
+// breaking change.
+const (
+	accountHK hashKind = iota
+	assetHK
+	appHK
+	kvHK
+)
+
+// HashKindEncodingIndex defines the []byte offset where the hash kind is
+// encoded.
+const HashKindEncodingIndex = 4
```

Comment:
looks not used outside and logically appears to be an implementation detail -> must be not exported

Reply:
@algorandskiy Thanks for catching - I previously needed it to be exposed, but that's no longer the case.  Fixed in https://github.com/algorand/go-algorand/pull/4808/commits/351f9bec42aa591d4fbad69957924f2ab51a49f9.  Can you resolve if it looks good?